### DISCUSSION TITLE
[A-L] (2023/24) Foglio 3 - Esercizio 5

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/calcolo-delle-probabilita#153] [Esercizi - Bertini] [answered]
![Es 5](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/107944476/dfcd4b49-713f-49b5-88f8-d88f8839ab22)

**Comment by Elia-Belli** [comment, marked as answer]
- **(1)** Una funzione è una mappa da un insieme $S$ ad un insieme $S'$ tale che ogni elemento in $S$ è mappato ad un unico elemento in $S'$. Quindi possiamo pensarla come un'estrazione ordinata con rimpiazzo, dove la prima estrazione da $S'$ è l'elemento mappato al "primo" elemento in $S$ (immaginando di numerare gli elementi di $S$), quindi abbiamo $k^n$ possibili funzioni.
- **(2)** Una funzione iniettiva ha la proprietà che nessun elemento mappato ha la stessa immagine, quindi possiamo pensarla come un'estrazione ordinata senza rimpiazzo, ovvero abbiamo $\frac{k!}{(k-n)!}$ funzioni iniettive, con il vincolo che $n\leq k$.
- **(3)** Una funzione biunivoca è sia iniettiva che suriettiva, ovvero a tutti gli elementi in $S$ è mappato un elemento distinto in $S'$ e tutti gli elementi di $S'$ sono immagine di qualche elemento in $S$. Affinchè ciò avvenga va applicato il vincolo $k=n$ e possiamo pensarla come una permutazione degli elementi di $S'$, quindi abbiamo $n!$ funzioni biunivoche.

**Comment by IonutCicio** [comment]
4. Definiamo $A = \\{ f:S \to S' \mid \forall s' \in S', \exists s \in S : f(s) = s' \\}$, bisogna contare la cardinalità di A. Farlo direttamente risulta abbastanza complicato. Conviene fare il passaggio a complemento, per cui $|A| = |F| - |A^c|$, con $F = \\{ f:S \to S' \\}$ e $A^c = \\{ f:S \to S' \mid \exists s' \in S' : \forall s \in S : f(s) \neq s' \\}$

Dal calcolo di @Elia-Belli al punto 1. sappiamo che $|F| = k^n$, ora si tratta solo di calcolare la cardinalità di $A^c$

Per farlo, prefissiamo il primo elemento di $S'$ come *non usato*, e risulta che ci sono $(k - 1)^n$ possibili funzioni in cui il primo elemento non è usato (per il primo elemento di $S$ ho $(k - 1)$ scelte in $S'$, anche per il secondo etc...). Per decidere quale elemento di $S$ non usare ho $\binom{k}{1}$ scelte.

Il problema è che l'intersezione fra l'insieme in cui il primo elemento non è usato e l'insieme in cui il secondo elemento non è usato **non è vuota**. Bisogna sottrare le intersezioni a due a due.

Si verifica abbastanza facilmente il numero di elementi nell'intersezione è $\binom{k}{2} (k - 2)^n$ perché levati il primo e il secondo elemento, per ogni elemento $s_1, s_2, ..., s_n \in S$ ho $(k - 2)$ scelte. Il numero delle intersezioni due a due è $\binom{k}{2}$

Ora bisogna aggiungere le intersezioni tre a tre etc.... per cui si ha che la cardinalità di $A^c$ è

$$
|A^c| = \binom{k}{1}(k - 1)^n - \binom{k}{2}(k - 2)^n + \binom{k}{3}(k - 3)^n - ... + \binom{k}{k}(k - k)^n =
\sum_{i = 1}^{k} (-1)^{i + 1}\binom{k}{i}(k - i)^n
$$

Ora abbiamo che 

$$
|A| = |F| - |A^c| = k^n - \sum_{i = 1}^{k} (-1)^{i + 1}\binom{k}{i}(k - i)^n = k^n + \sum_{i = 1}^{k} (-1)^i\binom{k}{i}(k - i)^n
$$

Se nel termine della sommatoria poniamo $i = 0$ otteniamo $1 \cdot 1 \cdot k^n$ che corrisponde al termine fuori dalla somma, quindi possiamo far partire la sommatoria da $0$ e l'evare il termine $k^n$

$$
|A| = \sum_{i = 0}^{k} (-1)^i\binom{k}{i}(k - i)^n
$$

**Comment by CiottoloMaggico** [comment]
Soluzione primi 3 punti analoga @Elia-Belli.
Propongo soluzione all'esercizio 4 analoga a @CuriousCI con procedimento simile. 
![image](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/79638284/6714633b-43f5-4555-8d9d-db50797aeead)
